test/mock_sdk.dart

```diff
@@ -160,6 +160,10 @@ class Future<T> {
   factory Future.delayed(Duration duration, [T computation()]) => null;
   factory Future.value([value]) => null;
   static Future wait(List<Future> futures) => null;
+  Future<T> catchError(Function onError, {bool test(Object error)});
+  Future<T> whenComplete(action());
+  Future<T> then(Future<T> onValue(T value), { Function onError });
+  // Original is Future<S> then<S>(FutureOr<S> onValue(T value), { Function onError });
```

Comment:
I was confused and thought that you had changed from the "original" definition to the new (even though you clearly added the commented out line a well). Adding those three methods is fine, but the definitions should match what is in the SDK. If that means that we also need to add the definition of `FutureOr`, then we should do so.